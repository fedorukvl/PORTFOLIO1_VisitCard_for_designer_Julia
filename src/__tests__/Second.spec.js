import React from "react";
import { Second } from "../Components/index.js";
import { shallow } from "enzyme";

describe("Tests of <Second/> component", () => {
	let secondComponent;
	beforeEach(() => {
		secondComponent = shallow(<Second />);
	});
	afterEach(() => {
		secondComponent = null;
	});
	describe("First render of <Second/> Component", () => {
		it("Renders properly", () => {
			expect(secondComponent).not.toBeNull();
			expect(secondComponent.debug()).toMatchSnapshot();
		});
	});

	describe("<Second/> component renders elements", () => {
		it("Renders elements properly", () => {
			expect(secondComponent.debug()).toMatchSnapshot();
		});

		it("Renders elements properly (checking divs)", () => {
			expect(secondComponent.find(".rightDiv")).toHaveLength(1);
			expect(secondComponent.find(".leftDiv")).toHaveLength(1);
		});
	});

	describe("Element`s children of <Second/> component render properly", () => {
		it(".rightDiv have children", () => {
			expect(secondComponent.find(".rightDiv").children()).toHaveLength(
				2
			);
		});

		it(".rightDiv have <img/> as children", () => {
			expect(secondComponent.find(".rightDiv").childAt(0).type()).toEqual(
				"img"
			);
		});
	});
});
