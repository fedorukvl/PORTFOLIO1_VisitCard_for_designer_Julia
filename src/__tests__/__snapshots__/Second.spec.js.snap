// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of <Second/> component <Second/> component renders elements Renders elements properly 1`] = `
"<div className=\\"basicCloseDiv\\">
  <div className=\\"leftDiv\\">
    <img src=\\"1.jpg\\" alt=\\"first concept\\" className=\\"firstImg\\" />
    <img src=\\"2.png\\" alt=\\"fourth concept\\" className=\\"secondImg\\" />
  </div>
  <div className=\\"rightDiv\\">
    <img src=\\"3.jpg\\" alt=\\"second concept\\" className=\\"thirdImg\\" />
    <img src=\\"4.jpg\\" alt=\\"third concept\\" className=\\"fourthImg\\" />
  </div>
</div>"
`;

exports[`Tests of <Second/> component First render of <Second/> Component Renders properly 1`] = `
"<div className=\\"basicCloseDiv\\">
  <div className=\\"leftDiv\\">
    <img src=\\"1.jpg\\" alt=\\"first concept\\" className=\\"firstImg\\" />
    <img src=\\"2.png\\" alt=\\"fourth concept\\" className=\\"secondImg\\" />
  </div>
  <div className=\\"rightDiv\\">
    <img src=\\"3.jpg\\" alt=\\"second concept\\" className=\\"thirdImg\\" />
    <img src=\\"4.jpg\\" alt=\\"third concept\\" className=\\"fourthImg\\" />
  </div>
</div>"
`;
